cdc/log.cc

```diff
@@ -1084,7 +1103,7 @@ struct process_row_visitor {
         // See `set_visitor`, `udt_visitior`, and `map_or_list_visitor` below.
         struct collection_visitor {
             bool _is_column_delete = false;
-            std::vector<managed_bytes_view> _deleted_keys;
+            std::vector<managed_bytes_view_opt> _deleted_keys;
```

Comment:
I'm afraid I don't follow the details here. If this vector is supposed to hold keys, why is it "opt"? How can the key (not the value) be null?

Reply:
If the data comes from a set, which shares the internal representation with a list, these could be opt (but never null).

We could de-opt them prior to adding them here. @kbr- any preference?